app/src/main/java/com/github/palFinderTeam/palfinder/ProfileActivity.kt

```diff
@@ -79,11 +83,57 @@ class ProfileActivity : AppCompatActivity() {
 
         viewModel.profile.observe(this) {
             when(it) {
-                is Response.Success -> injectUserInfo(it.data)
+                is Response.Success -> {
+                    injectUserInfo(it.data)
+                    bindFollow(null, it.data)
+                }
+                is Response.Loading -> Toast.makeText(applicationContext, "Fetching",  Toast.LENGTH_LONG).show()
+                is Response.Failure -> Toast.makeText(applicationContext, it.errorMessage, Toast.LENGTH_LONG).show()
+            }
+        }
+    }
+
+    /**
+     * binds the follow/unfollow button
+     */
+    private fun bindFollow(view: View?, profileViewed: ProfileUser) {
+        val followButton = findViewById<Button>(R.id.button_follow_profile)
+        viewModel.logged_profile.observe(this) {
+            when(it) {
+                is Response.Success -> {
+                   followSystem(it.data, profileViewed, followButton)
+                }
                 is Response.Loading -> Toast.makeText(applicationContext, "Fetching",  Toast.LENGTH_LONG).show()
                 is Response.Failure -> Toast.makeText(applicationContext, it.errorMessage, Toast.LENGTH_LONG).show()
```

Comment:
These messages are good for now but should definitely be removed in the final product.